Cleanup. (#3038)
wireapp/wire-server: docs/src/understand/single-sign-on/trouble-shooting.md

(trouble-shooting-faq)=

```{contents}
:depth: 2

> ```

Author: @arthurwolf you have used the same idiom in https://raw.githubusercontent.com/wireapp/wire-server/develop/docs/src/understand/single-sign-on/understand/main.md with much better success.  what am i doing wrong here?

(it's not important, only solve it if it's easy, otherwise we can keep it as it is now.  thanks!)

# Trouble shooting & FAQ

## Reporting a problem with user provisioning or SSO authentication

In order for us to analyse and understand your problem, we need at least the following information up-front:

- Have you followed the following instructions?
    - {ref}`FAQ <trouble-shooting-faq>` (This document)
    - [Howtos](https://docs.wire.com/how-to/single-sign-on/index.html) for supported vendors
    - [General documentation on the setup flow](https://support.wire.com/hc/en-us/articles/360001285718-Set-up-SSO-externally)

> - Which vendor (and product if /a) are you using (octa, azure, centrica, other (which one)?)

Reviewer: ```suggestion
- Which vendor (or product) are you using (octa, azure, centrica, other (which one)?)
```
I don't understand `... if /a`

Author: "if available".  but ok, i'll change it.

- Team ID (looks like eg. `2e9a9c9c-6f83-11eb-a118-3342c6f16f4e`, can be found in the team management app)
- User ID of the account that has the problem (alternatively: handle, email address)
- What do you expect to happen?

>     - eg.: "I enter login code, authenticate successfully against IdP, get redirected, and see the wire landing page."

Reviewer: ```suggestion
    - e.g.: "I enter login code, authenticate successfully against IdP, get redirected, and see the wire landing page."
```

Author: i don't like this, it's correct but awkward...  but ok!

- What does happen instead?
    - Screenshots
    - Copy the text into your report where applicable in addition to screenshots (for automatic processing).

>     - eg.: "instead of being logged into wire, I see the following error page: ..."

Reviewer: ```suggestion
    - e.g.: "instead of being logged into wire, I see the following error page: ..."
```

- Screenshots of the Configuration (both SAML and SCIM, as applicable), including, but not limited to:
    - If you are using SAML: SAML IdP metadata file
    - If you are using SCIM for provisioning: Which attributes in the User schema are mapped?  How?
- If you have successfully authenticated on your IdP and are
  redirected into wire, and then see a white page with an error
  message that contains a lot of machine-readable info: copy the full
  message to the clipboard and insert it into your report.  We do not
  log this information for privacy reasons, but we can use it to
  investigate your problem.  (Hint, if you want to investigate

>   yourself: it's base64 encoded!

Reviewer: ```suggestion
  yourself: it's base64 encoded!)
```

### notes for wire support / development

Not officially supported IdP vendors may work out of the box, as we
are requiring a minimum amount of SAML features.

If there are problems: collect the metadata xml and an authentication
response xml (either from the browser http logs via a more technically
savvy customer; or from the "white page with an error" mentioned
above.

https://github.com/wireapp/saml2-web-sso supports writing [unit vendor
compatibility
tests](https://github.com/wireapp/saml2-web-sso/blob/ff9b9f445475809d1fa31ef7f2932caa0ed31613/test/Test/SAML2/WebSSO/APISpec.hs#L266-L329)

> against that response value.  once that test passes, it should all

Reviewer: ```suggestion
against that response value.  Once that test passes, it should all
```

work fine.

## Can I use SCIM without SAML?

Yes.  Scim is a technology for onboarding alternative to the team management app, and can produce both user accounts authenticated via SAML or via email and password.  (Phone may or may not work, but is not officially supported.)

> How does it work?  -- Make sure your team has no SAML IdPs registered.  Set up your SCIM peer to provision users with valid email addresses as `externalIds`.  Newly provisioned users will be created in status `PendingInvitation`, and an invitation email will be sent.  From here on out, the flow is exactly the same as if you had added the user to your team in the team management app.

Reviewer: ```suggestion
How does it work?  Make sure your team has no SAML IdPs registered.  Set up your SCIM peer to provision users with valid email addresses as `externalIds`.  Newly provisioned users will be created in status `PendingInvitation`, and an invitation email will be sent.  From here on out, the flow is exactly the same as if you had added the user to your team in the team management app.
```

Upcoming features:
- support for the `emails` field in the scim user record (so you can choose non-email `externalId` values).
- flexible mapping between any number of SAML IdPs and any number of SCIM tokens in team management.

## Can I use SAML without SCIM?

Yes, but this is not recommended.  User (de-)provisioning requires more manual work without SCIM, and some of the account information cannot be provisioned at all via SAML.

## Can I use the same SSO login code for multiple teams?

Most SAML IdP products allow you to register arbitrary many apps for

> arbitrary many teams, by using as many different entity Ids.  This is

Reviewer: ```suggestion
arbitrary many teams, by using a different entity Id for each app/team.  This is
```

currently supported out of the box.

> If you don't have this option, ie. you need to serve two teams with an

Reviewer: ```suggestion
If you don't have this option, i.e. you need to serve two teams with an
```

> IdP that has only one entityId, please keep reading and/or contact

Reviewer: ```suggestion
IdP that has only one entity ID, please keep reading and/or contact
```

customer support.

### The long answer

> Some SAML IdP vendors do not allow to set up fresh entityIDs (issuers)

Reviewer: ```suggestion
Some SAML IdP vendors do not allow to set up fresh entity IDs (issuers)
```

for fresh apps; instead, all apps controlled by the IdP are receiving
SAML credentials from the same issuer.

> In the past, wire has used the a tuple of IdP issuer and 'NameID'

Reviewer: ```suggestion
In the past, wire has used a tuple of IdP issuer and 'NameID'
```
